src/workers/tweetWorker.ts

```diff
@@ -31,29 +32,54 @@ export const generateAndTweet = async (prompt: string, newsPrompt: string) => {
       return;
     }
 
+    const handles =
+      newsHandles && newsHandles.length > 0 ? newsHandles : [newsUsername];
+
     // check if it's time to include latest news in the tweet or not
     const newsCacheKey = getCacheKey(`lastnewstime`);
     let timestamp = await cacheClient.get(newsCacheKey);
     let gptResponse: ChatCompletion.Choice;
+
     if (!timestamp || isOlderThanXHours(parseInt(timestamp), 3)) {
       console.log("---------NEWS TIME-----------");
-      const userProfile = await getUserProfileByUsername(newsUsername);
-      // fetch last post from MarioNawfal
-      const tweets = await twitterClient.v2.userTimeline(userProfile.id, {
-        max_results: 5,
-      });
-      const news = tweets?.data?.data[0];
+
+      const fetchTweetsFromHandle = async (handle: string) => {
+        try {
+          const userProfile = await getUserProfileByUsername(handle);
+          const tweets = await twitterClient.v2.userTimeline(userProfile.id, {
+            max_results: 5,
+          });
+          return tweets?.data?.data[0]?.text;
+        } catch (error) {
+          console.error(`Error fetching tweets from ${handle}:`, error);
+          return null;
+        }
+      };
+
+      let newsContent: string | null = null;
+      for (const handle of handles) {
```

Comment:
let's select one randomly? otherwise it will mostly fetch tweets from 1st handle itself

Reply:
or shuffle, whatever is best

Reply:
shuffle sounds good